### ISSUE TITLE
[BUG] if attribute value is a dict in update/append request, the error is not controlled

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#1217]
In update/append request if attribute value is a  dict and a key has a dot, ex: `{"rt.ty": "5678"}` the error is not controlled correctly.

### Request:
```
POST http://<host>:<port>/v2/entities/room_1
  Content-Type: application/json
  Fiware-Service: test_update_attribute_value_special
  Fiware-ServicePath: /test
  {"temperature_0": {"rt.ty": "5678"}}
```

### Response:
```
 status code: 500
  date: Thu, 10 Sep 2015 10:34:27 GMT
  content-length: 52
  content-type: application/json
  {"error":"Internal Server Error","description":"""}
```

**Comment by kzangeli** [comment]
The value for the attribute "temperature_0" is a compound, {"rt.ty": "5678"}.
Two possibilities: **400 Bad Request** if we don't allow dots in the tag (rt.ty), **200 OK** if we do ...
Never a 500, of course ...

**Comment by fgalan** [comment]
MongoDB has some problems to use the dot character (".") as key name. Probably the 500 Error is generated by MongoDB due to this.

In order to assure it, do a simple test: just replace "." by some other character (e.g. the "+" symbol) and check if it works or not.

**Comment by kzangeli** [comment]
Func test created (creation of entity) and no problem at all ...

    --SHELL-INIT--
    dbInit CB
    brokerStart CB

    --SHELL--

    #
    # 01. Create entity E1 with attr A1 with a compound value with a dot in the tag
    # 02. GET /v2/entities/E1
    #


    echo "01. Create entity E1 with attr A1 with a compound value with a dot in the tag"
    echo "============================================================================="
    payload='{ "temperature_0": {"rt.ty": "5678"} }'
    orionCurl --url /v2/entities/E1 --payload "$payload" --json
    echo
    echo


    echo "02. GET /v2/entities/E1"
    echo "======================="
    orionCurl --url /v2/entities/E1 --json
    echo
    echo


    --REGEXPECT--
    01. Create entity E1 with attr A1 with a compound value with a dot in the tag
    =============================================================================
    HTTP/1.1 201 Created
    Content-Length: 0
    Location: /v2/entities/E1
    Date: REGEX(.*)



    02. GET /v2/entities/E1
    =======================
    HTTP/1.1 200 OK
    Content-Length: 44
    Content-Type: application/json
    Date: REGEX(.*)

    {
        "id": "E1",
        "temperature_0": {
            "rt.ty": "5678"
        }
    }

**Comment by fgalan** [comment]
This issue is related with the JSON format for representing entities/attributes/metadata, which is currently under discussion. Thus, it has been deferred to milestone 0.26.0.

**Comment by iariasleon** [comment]
Appendding a metadata if the entitiy does not exists, it is created:
```
{ "_id" : { "id" : "room_4", "servicePath" : "/test" }, "attrNames" : [ "temperature_0" ], "attrs" : { "temperature_0" : { "type" : "", "creDate" : 1442303878, "modDate" : 1442303878, "value" : { "rt.ty" : "5679" } } }, "creDate" : 1442303878, "modDate" : 1442303878 }
```
But if we try to update this metadata occurs the error above mentioned.

**Comment by raconte** [comment]
#### CASE 

### Create an entity (POST)--> the same as the first example shown 
```
(curl localhost:1026/v2/entities/apartment_5 -s -S --header Content-Type:application/json --header 'Accept: application/json' -d @-| python -mjson.tool)<< EOF
{
	"temperature_2": {
		"rt.ty": "5678"
	}
}
EOF
```
### Response
``` 
{
    "description": "no 'value' for ContextAttribute without keyValues",
    "error": "BadRequest"
}
```
#### CASE

Create the entity --> Update

### Create an entity (POST)
```
(curl localhost:1026/v2/entities --header Content-Type:application/json --header 'Accept: application/json' -d @- | python -mjson.tool)<< EOF
{
	"type": "house",
	"id": "apartment_3",	
	"temperature_2": {
		"value": 40
	}
}
EOF
```
### Update/append request (PUT)
```
(curl localhost:1026/v2/entities/apartment_3 -s -S --header Content-Type:application/json --header 'Accept: application/json'  -X PUT -d @-| python -mjson.tool)<< EOF
{
	"temperature_2": {
		"rt.ty": "5678"
	}
}
EOF
```

### Response
```
{
    "description": "no 'value' for ContextAttribute without keyValues",
    "error": "BadRequest"
}
```
Solved

**Comment by fgalan** [comment]
Issue fixed in PR #1739. Please @iariasleon have check if it has been finaly solved.

**Comment by iariasleon** [comment]
LGTM
```
{"id":"room_1","type":null,"temperature_0":{"type":"date","value":{"rt.ty":"5678"},"metadata":{}}}
```
